Sources/FluentKit/Properties/OptionalGroup.swift

```diff
@@ -0,0 +1,82 @@
+extension Fields {
+    public typealias OptionalGroup<Value> = OptionalGroupProperty<Self, Value>
+        where Value: Fields
+}
+
+@propertyWrapper @dynamicMemberLookup
+public final class OptionalGroupProperty<Model, Value>
+    where Model: FluentKit.Fields, Value: FluentKit.Fields
+{
+    public let key: FieldKey
+    public var value: Value?
+
+    public var projectedValue: OptionalGroupProperty<Model, Value> {
+        return self
+    }
+
+    public var wrappedValue: Value? {
+        get {
+            return self.value
+        }
+        set {
+            self.value = newValue
+        }
+    }
+
+    public init(key: FieldKey) {
+        self.key = key
+        self.value = .init()
+    }
+
+    public subscript<Property>(
+         dynamicMember keyPath: KeyPath<Value, Property>
+    ) -> NestedProperty<Model, Property>
+        where Property: PropertyProtocol
+    {
+        .init(prefix: [self.key], property: self.value![keyPath: keyPath])
+    }
+}
+
+
+extension OptionalGroupProperty: PropertyProtocol { }
+
+extension OptionalGroupProperty: AnyProperty {
+    public var nested: [AnyProperty] {
+        self.value!.properties
+    }
+
+    public var path: [FieldKey] {
+        [self.key]
+    }
+
+    public var requiresUnexceptedNilCheck: Bool {
+        return true
+    }
+
+    public func input(to input: inout DatabaseInput, chain: [AnyProperty]) {
+        if let values = self.value?.input(chain: chain+[self]).values, !values.isEmpty  {
+            input.values[self.key] = .dictionary(values)
+        }
+    }
+
+    public func output(from output: DatabaseOutput, chain: [AnyProperty]) throws {
+        do {
+            try self.value?.output(from: output.nested(self.key), chain: chain+[self])
```

Comment:
> If Fields contain no required fields we cannot determine if OptionalGroup should be nil or not resulting it in never being nil. 

This feels like it has the potential to cause problems or confusion. For example, if you have a model with these fields:

```swift
struct Foo: Fields {
    @OptionalField var bar: Int?
    @OptionalField var baz: String?
}

@OptionalGroup var foo: Foo?
```

Say you save a new model with `foo = `nil`:

```swift
let a = Model()
a.foo = nil
try a.save(on: db).wait()
```

When you fetch this model, `foo` will no longer be `nil` because all of `Foo`'s fields are optional so no "unexpectedNil" error was thrown. 

```swift
let b = try Model.find(a.id!, on: db).wait()
print(b.foo) // Expected nil, but it is Foo
print(b.foo?.bar) // nil
print(b.foo?.baz) // nil
```

Reply:
I don't think thats confusing.
We can just point that out in the wiki.
Having an @OptionalGroup with all fields beeing optional doesn't make any sense design wise.